backend/hitas/views/building.py

```diff
@@ -50,12 +51,21 @@ class BuildingViewSet(HitasModelViewSet):
     serializer_class = BuildingSerializer
     model_class = Building
 
+    def perform_destroy(self, instance: Building) -> None:
+        if instance.apartments.exists():
+            raise ModelConflict("Cannot delete a building with apartments.", error_code="apartments_on_building")
+
+        super().perform_destroy(instance)
+
     def get_queryset(self):
         hc_id = self._lookup_model_id_by_uuid(HousingCompany, "housing_company_uuid")
         re_id = self._lookup_model_id_by_uuid(RealEstate, "real_estate_uuid", housing_company_id=hc_id)
 
         return (
             Building.objects.filter(real_estate__id=re_id)
+            .prefetch_related(
+                Prefetch("apartments", Apartment.objects.only("uuid")),
+            )
```

Comment:
This prefetch is adding one extra apartments db query for each apartment the building has 😬 
Why should we prefetch the Apartment uuid's anyway? 🤔

Reply:
Prefetch only adds [one query for all apartments](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#:~:text=Returns%20a%20QuerySet%20that%20will%20automatically%20retrieve%2C%20in%20a%20single%20batch%2C%20related%20objects%20for%20each%20of%20the%20specified%20lookups.).

The idea here was have the apartments already fetched in the wueryst, so that `instance.apartments.exists():` during DELETE wouldn't make another database query, but I think it doesn't make sense in hindsight since it's only needed for the DELETE endpoint, and the number of queries is the same either way, so I'll remove it.

Reply:
This change is in the latest commit.